Update nexus

ironpinguin opened the pull request:
###### Motivation for this change
Update and Fix the existing Nexus Repository Server. Also Add a module for the service 

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Ma27 reviewed (commented):
I like the changes overall (and I'm glad that you do Nix :P)

ironpinguin commented on pkgs/development/tools/repository-managers/nexus/nexus-bin.patch:
```diff
@@ -0,0 +1,60 @@
++++ nexus	2017-08-31 12:41:01.271475645 +0200
+@@ -1,4 +1,4 @@
+-#!/bin/sh
++#!/nix/store/axldd4qjamb2c1lgi7d6k3s51qyy4610-bash-4.4-p12/bin/sh
```
Ok take a look on it.

ironpinguin commented on pkgs/development/tools/repository-managers/nexus/default.nix:
```diff
@@ -1,29 +1,48 @@
     license = licenses.epl10;
     platforms = platforms.all;
-    maintainers = [ maintainers.aespinosa ];
+    maintainers = [ maintainers.aespinosa maintainers.ironpinguin ];
```
I will change it

NeQuissimus commented on nixos/tests/nexus.nix:
```diff
@@ -0,0 +1,32 @@
+# verifies:
```
I know where you copied this from :D

ironpinguin replied:
:-D

ironpinguin replied:
Change it now to more reasonable comments ;-)

NeQuissimus commented on pkgs/development/tools/repository-managers/nexus/default.nix:
```diff
@@ -1,29 +1,48 @@
   src = fetchurl {
-    url = "https://sonatype-download.global.ssl.fastly.net/nexus/oss/nexus-${version}-bundle.tar.gz";
-    sha256 = "1k3z7kwcmr1pxaxfnak99fq5s8br9zbqbfpyw1afi86ykkph4g5z";
+    url = "https://sonatype-download.global.ssl.fastly.net/nexus/3/nexus-${version}-mac.tgz";
```
The macOS bundle works cross-platform?!

ironpinguin replied:
Yes because there are only two diffrents:
1. in .install4j folder there comes also a jre.bundle in mac style (if no local jre is installed)
2. Additional checks for jre in mac style (include fallback to jre.bundle) in startup shell script

And both has no side effects on unix systems (expect the extra 167M for the jre bundle)

I test it on my nixos box.

NeQuissimus commented on nixos/tests/nexus.nix:
```diff
@@ -0,0 +1,32 @@
+
+    print $server->execute("sudo -u nexus groups");
+    $server->mustSucceed("sudo -u nexus groups | grep nexus | grep users");
+  '';
```
Could you add a test that actually checks that the Nexus interface is up?

ironpinguin replied:
I can do it. First i add it but got some memory error in the test kvm. But can also come from the situation that i run the nixos in a virtualbox :-(

NeQuissimus commented on nixos/modules/services/web-apps/nexus.nix:
```diff
@@ -0,0 +1,100 @@
+
+        ln -sf ${cfg.home} /run/sonatype-work
+
+        chown -R ${cfg.user} ${cfg.home}
```
Won't make much of a difference but should the chown also take into account the group?

ironpinguin replied:
Oh yes i will do it.

NeQuissimus commented:
Looks pretty good, just a few suggestions here and there

globin commented:
Commits should be prefixed with `nexus: ` and the version bump be of the form `nexus: a.b.c -> x.y,z`.

Please also try to squash the commits into logical entities.

ironpinguin committed:
nexus: 2.12.0-01 -> 3.5.1-02

Update Nexus to version 3.5.1-02 and fix startup stript to run in nix
environment.

ironpinguin committed:
nexus: Add module for nexus.

Add also myself as maintainer
Add simple test of the nexus service

ironpinguin commented:
@globin now ok?
